From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Himanshu Rawat <rwt@google.com>
Date: Tue, 11 Jul 2023 22:20:54 +0000
Subject: [PATCH] Revert "Ignore additional encrypt change events when link is
 auth & encrypted"

This reverts commit 4b0308e5c6eb558881926fcfc0a8b05f0be89766.

Reason for revert: Prevents CTKD on upgrading security on an already encrypted link

Bug: 287300872
Change-Id: I17bba54af12458fd76973f09e6e1b1a44192a689
---
 system/stack/btm/btm_sec.cc | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/system/stack/btm/btm_sec.cc b/system/stack/btm/btm_sec.cc
index 2d87034024..bd5501cb84 100644
--- a/system/stack/btm/btm_sec.cc
+++ b/system/stack/btm/btm_sec.cc
@@ -3458,13 +3458,6 @@ void btm_sec_encrypt_change(uint16_t handle, tHCI_STATUS status,
   if (status == HCI_SUCCESS) {
     if (encr_enable) {
       if (p_dev_rec->hci_handle == handle) {  // classic
-        if ((p_dev_rec->sec_flags & BTM_SEC_AUTHENTICATED) &&
-            (p_dev_rec->sec_flags & BTM_SEC_ENCRYPTED)) {
-          LOG_INFO(
-              "Link is authenticated & encrypted, ignoring this enc change "
-              "event");
-          return;
-        }
         p_dev_rec->sec_flags |= (BTM_SEC_AUTHENTICATED | BTM_SEC_ENCRYPTED);
         if (p_dev_rec->pin_code_length >= 16 ||
             p_dev_rec->link_key_type == BTM_LKEY_TYPE_AUTH_COMB ||
-- 
2.34.1

